### ISSUE TITLE
HTML Compression does not work with SSL

### ISSUE THREAD

**Issue by petehannam** [websharks/zencache#348]
Hi,

We've been running Quick Cache Pro successfully for a while now.  Over the weekend we moved the site from HTTP to HTTPS (i.e. added an SSL cert to the server).  

With Quick Cache's HTML compression turned off, everything works fine.  However, we found the site broke if we enabled any of the following options.  By 'broke' I mean that the CSS and/or JS resources didn't load.

- "Combine CSS from <head> and <body> into fewer files"
- "Combine JS from <head> into fewer files"

Thanks,
Pete

**Comment by raamdev** [comment]
### Related GitHub Issue

Noting here that there's another GitHub Issue (https://github.com/websharks/zencache/issues/413) related to this one. I'm closing this GitHub issue in favor of the other one, as there's a bit more insight in that issue regarding where the problem may be.

---

@petehannam Could you leave a comment on https://github.com/websharks/zencache/issues/413 and let me know if you were using the CloudFlare Minify & Compression features while also trying to use the HTML Compressor? If you were, that might explain the conflict.

**Comment by raamdev** [comment]
@petehannam Thank you for your patience on this issue. We've identified a fix that we believe resolves the problem you described here. The following will be released with the next ZenCache Pro release:
- **Bug Fix**: Fixed an SSL issue with the HTML Compressor that was causing problems in some hosting environments where the hosting server was incorrectly setting `$_SERVER['REQUEST_SCHEME']` to `http` even when the WordPress Site URL and Home URL were set to use `https://`. As a result of this improper server configuration, the combined CSS/JS files generated by the HTML Compressor were being served over HTTP even when a site was configured to use HTTPS. This release applies a workaround for this improper server configuration that no longer looks at `$_SERVER['REQUEST_SCHEME']`. See [Issue #413](https://github.com/websharks/zencache/issues/413) and [Issue #73](https://github.com/websharks/html-compressor/issues/73).
